crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -619,6 +622,10 @@ pub struct AuthorityEpochTables {
     /// Execution time observations for congestion control.
     pub(crate) execution_time_observations:
         DBMap<(u64, AuthorityIndex), Vec<(ExecutionTimeObservationKey, Duration)>>,
+
+    /// Transactions that have been rejected through mysticeti.
+    // TODO: Propagate the reason for rejection.
+    pub(crate) mysticeti_rejected_transactions: DBMap<ConsensusTransactionPosition, ()>,
```

Comment:
Can we use a BTreeMap instead? I was thinking that we may only need to keep (consensus position, digest) -> final status in memory. Basically in the rare cases where in memory statuses are lost on restart and this impacts reaching finality on transaction driver, the transaction driver can always retry submitting the transaction.

Reply:
Could someone DoS it and lead to OOM?

Reply:
I will write this up in more details, but basically I think we can just maintain an in-memory map of (epoch, consensus position) -> consensus status. The statuses will be cleaned up as consensus rounds advance. If a transaction driver is requesting status from an old position, it should just be told so. The transaction driver should be able to check if the transaction has executed or is executable locally. If the transaction has not executed and is still executable, it can resubmit the transaction. This would reduce disk write and potentially make epoch changes easier too.

Reply:
Are you saying that there is a K such that if the current consensus round is K bigger than the position, that position information can be deleted and forgotten?

Reply:
Yes!